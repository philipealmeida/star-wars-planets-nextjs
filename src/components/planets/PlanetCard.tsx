import Link from 'next/link';
import type { Planet } from '@/lib/types';
import styles from '@/styles/components/planets/PlanetCard.module.css';

interface PlanetCardProps {
  planet: Planet;
}

export default function PlanetCard({ planet }: PlanetCardProps) {
  const planetId = planet.url.split('/').filter(Boolean).pop();
  
  return (
    <Link href={`/planets/${planetId}`} className={styles.card}>
      <h2>{planet.name}</h2>
      <div className={styles.details}>
        <p>Climate: {planet.climate}</p>
        <p>Terrain: {planet.terrain}</p>
        <p>Population: {planet.population}</p>
      </div>
    </Link>
  );
}