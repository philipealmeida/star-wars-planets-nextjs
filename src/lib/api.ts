const SWAPI_BASE_URL = 'https://swapi.dev/api';

export const fetchPlanets = async (page: number = 1, search?: string) => {
  const searchParams = new URLSearchParams({
    page: page.toString(),
    ...(search && { search }),
  });
  
  const response = await fetch(
    `${SWAPI_BASE_URL}/planets/?${searchParams.toString()}`
  );
  
  if (!response.ok) {
    throw new Error('Failed to fetch planets');
  }
  
  return response.json();
};

export const fetchPlanetById = async (id: string) => {
  const response = await fetch(`${SWAPI_BASE_URL}/planets/${id}`);
  
  if (!response.ok) {
    throw new Error('Failed to fetch planet details');
  }
  
  return response.json();
};