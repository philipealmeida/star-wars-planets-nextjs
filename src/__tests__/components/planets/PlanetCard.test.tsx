import { Planet } from '@/lib/types';
import { render, screen } from '@testing-library/react';
import PlanetCard from '@/components/planets/PlanetCard';

const mockPlanet: Planet = {
    name: 'Tatooine',
    climate: 'Arid',
    terrain: 'Desert',
    population: '200000',
    url: 'https://swapi.dev/api/planets/1/',
    diameter: '10465',
    gravity: '1 standard',
    surface_water: '1',
    rotation_period: '',
    orbital_period: '',
    residents: [],
    films: [],
    created: '',
    edited: ''
};

const renderComponent = (props: { planet: Planet }) => {
  return render(<PlanetCard {...props} />);
};

describe('PlanetCard', () => {
  it('matches snapshot', () => {
    const { container } = renderComponent({ planet: mockPlanet });
    expect(container).toMatchSnapshot();
  });

  it('displays planet information correctly', () => {
    renderComponent({ planet: mockPlanet });
    
    expect(screen.getByText('Tatooine')).toBeInTheDocument();
    expect(screen.getByText('Climate: Arid')).toBeInTheDocument();
    expect(screen.getByText('Terrain: Desert')).toBeInTheDocument();
    expect(screen.getByText('Population: 200000')).toBeInTheDocument();
  });

  it('creates correct link with planet ID', () => {
    renderComponent({ planet: mockPlanet });
    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', '/planets/1');
  });
});