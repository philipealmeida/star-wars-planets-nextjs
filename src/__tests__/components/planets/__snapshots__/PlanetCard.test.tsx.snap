// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetCard matches snapshot 1`] = `
<div>
  <a
    class="card"
    href="/planets/1"
  >
    <h2>
      Tatooine
    </h2>
    <div
      class="details"
    >
      <p>
        Climate: 
        Arid
      </p>
      <p>
        Terrain: 
        Desert
      </p>
      <p>
        Population: 
        200000
      </p>
    </div>
  </a>
</div>
`;
